cpp/pybind/CMakeLists.txt

```diff
@@ -102,6 +102,13 @@ set_target_properties(${PACKAGE_NAME} PROPERTIES
                       FOLDER "Python"
                       LIBRARY_OUTPUT_DIRECTORY "${PYTHON_COMPILED_MODULE_DIR}"
                       ARCHIVE_OUTPUT_DIRECTORY "${PYTHON_COMPILED_MODULE_DIR}")
+# Include additional libraries that may be absent from the user system
+# eg: libc++.so and libc++abi.so (needed by filament) for Linux.
+# libc++.so is a linker script including libc++.so.1 and libc++abi.so, so append 1 to libc++.so
+set(PYTHON_EXTRA_LIBRARIES "")
+if (BUILD_GUI AND $<PLATFORM_ID:Linux>)
```

Comment:
Seems like `$<PLATFORM_ID:Linux>` might not be working on some system. I tried:
```
if (BUILD_GUI AND $<PLATFORM_ID:Linux>)
    message(FATAL_ERROR "here")
    list(APPEND PYTHON_EXTRA_LIBRARIES ${CPP_LIBRARY}.1 ${CPPABI_LIBRARY})
endif()
```
and cmake didn't crash. Also I couldn't find the copied `libc+.so` files in the wheel package. Could we take a second look at this?

We can add a `if (LINUX)` in the top-level CMake. Similar to https://github.com/google/filament/blob/3d121eb873fd138de10a458a65973d1b4af8a84c/CMakeLists.txt#L82, we can have:

```
if (UNIX AND NOT APPLE)
    set(LINUX TRUE)
endif()
```

We have a bunch of `if (UNIX AND NOT APPLE)` in the code base, so this can simplify them as well.

Reply:
Thanks for catching that. I assumed the generator expression would be evaluated in the `if()`, but they are only evaluated during generation, not during configuration.
I found a cleaner method to check the platform, with the `CMAKE_SYSTEM_NAME` variables
https://gitlab.kitware.com/cmake/community/-/wikis/doc/tutorials/How-To-Write-Platform-Checks